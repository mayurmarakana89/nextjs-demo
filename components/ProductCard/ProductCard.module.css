.root {
    max-width: 340;
    display: flex;
    justify-content: center;
    max-height: 300;
}
  
.media {
    text-align: center;
  }

.title {
    height: 40;
    margin-top: 20;
}
