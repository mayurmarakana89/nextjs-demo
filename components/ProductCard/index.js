import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import Image from 'next/image';

const useStyles = makeStyles({
  root: {
    maxWidth: 340,
    display: "flex",
    justifyContent: "center",
    maxHeight: 300
  },
  media: {
    textAlign: 'center'
  },
  title: {
    height: 40,
    marginTop: 20
  },
});

const ProductCard = ({ product = {} }) => {
  if (!product) return <></>;

  const classes = useStyles();

  return (
    <Card className={classes.root} data-testid={`product-${product.name}`}>
        <CardContent>
            <div className={classes.media}>
                <Image
                    width={200}
                    height={200}
                    alt={product.name}
                    src={product.default_image_urls.main_image_url}
                />
            </div>
            <Typography className={classes.title} align="center" variant="body2" color="textSecondary">
                {product.name}
            </Typography>
        </CardContent>
    </Card>
  );
}
export default ProductCard;