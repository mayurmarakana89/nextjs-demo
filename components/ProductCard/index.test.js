// Test cases
// 0. matches snapshot
// 1. should render the component
// 2. should verify that the image component is rendered properly
// 3. should verify that product name as expected

import React from "react";
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from "@testing-library/react";
import ProductCard from "./";

jest.mock("next/config", () => {
    return () => ({
        images: {
            domains: ['cdn.mec.ca'],
        },
    });
});

jest.mock("next/image", () => {
    return () => <img alt="alt-text"></img>;
});

const stubProduct = {
    name: "test product",
    default_image_urls: {
        main_image_url: "https://cdn.mec.ca/medias/sys_master/fallback/fallback/8917710503966/5056608-STA00-fallback.jpg"
    }
};

afterEach(cleanup);

describe("ProductCard", () => {
    it('matches snapshot', () => {
        const tree = renderer
          .create(<ProductCard product={stubProduct} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

    it("should render the component", () => {
        render(<ProductCard  product={stubProduct} />);
        expect(screen.getByTestId(`product-${stubProduct.name}`)).toBeInTheDocument();
    });

    it("should verify that the image component is rendered properly", () => {
        render(<ProductCard  product={stubProduct} />);
        const imgComponent = screen.getByAltText("alt-text");
        expect(imgComponent).toBeInTheDocument();
    });

    it("should verify that product name as expected", () => {
        render(<ProductCard  product={stubProduct} />);
        const nameComponent = screen.getByText("test product");
        expect(nameComponent).toBeInTheDocument();
    });
});