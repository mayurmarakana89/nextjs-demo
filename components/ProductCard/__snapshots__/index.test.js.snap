// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="product-test product"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="makeStyles-media-2"
    >
      <img
        alt="alt-text"
      />
    </div>
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      test product
    </p>
  </div>
</div>
`;
